dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -62,34 +65,45 @@
   private static void setDetectedOsSystemProperties() {
     // System properties to select Netty dependencies through os-maven-plugin
     // Definition of the properties: https://github.com/trustin/os-maven-plugin
-    String nonAlphaNumericPattern = "[^a-z0-9]+";
+
+    System.setProperty("os.detected.name", osDetectedName());
+
+    System.setProperty("os.detected.arch", osDetectedArch());
+
+    System.setProperty("os.detected.classifier", osDetectedName() + "-" + osDetectedArch());
+  }
+
+  private static String osDetectedName() {
     String osNameNormalized =
-        System.getProperty("os.name", "generic")
-            .toLowerCase(Locale.ENGLISH)
-            .replaceAll(nonAlphaNumericPattern, "");
+        NON_ALPHA_NUMERIC_PATTERN
```

Comment:
I forgot about `CharMatcher` in Guava! Consider this:

```java
private static final CharMatcher ALPHA_NUMERIC = CharMatcher.inRange('a', 'z').or(CharMatcher.inRange('0', '9'));

String osNameNormalized =
    ALPHA_NUMERIC.retainFrom(System.getProperty("os.name", "generic").toLowerCase(ENGLISH));
```

A little simpler to do `retainFrom(foo)` than `matcher(foo).replaceAll("")`.

Reply:
`retainFrom` is nice. Updated.